tools-and-tests/tools/src/main/java/org/hiero/block/tools/days/subcommands/GcpLsCommand.java

```diff
@@ -0,0 +1,398 @@
+                r.localOnly.forEach(f -> System.out.println("        " + f));
+            }
+        }
+    }
+
+    private void printLsView(NodeResult r) {
+        // optional full ls-style view
+        if (!showLs) {
+            return;
+        }
+
+        System.out.printf("    LS for node 0.0.%d:%n", r.node);
+
+        if (r.gcpFiles != null && !r.gcpFiles.isEmpty()) {
+            System.out.println("      GCP files:");
+            r.gcpFiles.forEach(f -> System.out.println("        " + f));
+        } else {
+            System.out.println("      GCP files: (none)");
+        }
+
+        if (!skipLocalCompare) {
+            if (r.localFiles != null && !r.localFiles.isEmpty()) {
+                System.out.println("      Local files:");
+                r.localFiles.forEach(f -> System.out.println("        " + f));
+            } else {
+                System.out.println("      Local files: (none)");
+            }
+        }
+    }
+
+    private void printView(List<NodeResult> results) {
+        for (final NodeResult r : results) {
+            printDifferenceView(r);
+            printLsView(r);
+        }
+    }
+
+    private void validateInputs() {
+        if (endDate.isBefore(startDate)) {
+            throw new IllegalArgumentException("end-date must be >= start-date");
+        }
+        if (samples <= 0) {
+            throw new IllegalArgumentException("--samples must be > 0 (got " + samples + ")");
+        }
+        if (samples > MAX_SAMPLES) {
+            throw new IllegalArgumentException(String.format(
+                    "--samples must be <= %d to avoid overwhelming output (got %d)", MAX_SAMPLES, samples));
+        }
+    }
+
+    /**
+     * Check one node for a given date+hour:
+     *
+     *  - GCP: uses MainNetBucket.listHour(date),
+     *         then filters blobs for this node and record file extensions.
+     *  - Local: counts *.rcd / *.rcd.gz files in downloadedDaysDir for date+hour+node.
+     */
+    private NodeResult checkNodeForHour(LocalDate date, int hour, int node) {
+        List<String> gcpSigFiles = List.of();
+        long gcpSigCount = 0; // default to 0
+        long localSigCount = 0; // default to 0
+        long diff = 0; // default to 0
+
+        List<String> gcpOnly = List.of();
+        List<String> localOnly = List.of();
+        List<String> localRecordFiles = List.of();
+
+        try {
+            var blobs = mainNetBucket.listHour(date.toEpochDay());
+            final String hourToken = date + "T" + String.format("%02d", hour) + "_";
+
+            gcpSigFiles = blobs.stream()
+                    .map(b -> b.path())
+                    .filter(path -> path.contains(hourToken))
+                    .filter(this::isRecordName)
+                    .filter(path -> matchesNode(path, node))
+                    .map(this::extractFileName)
+                    .toList();
+
+            gcpSigCount = gcpSigFiles.size();
```

Comment:
this will almost always be 1 and will tell you there is a single record file not that there is a signature file.